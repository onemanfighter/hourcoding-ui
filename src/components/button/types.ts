type ButtonProps = {
  isDisabled?: boolean;
  size: 'small' | 'medium' | 'large';
  clickHandler?: () => void;
} & (ButtonWithText | ButtonWithIcon | ButtonWithIconAndText);

type ButtonWithText = {
  variant: 'contained' | 'outlined' | 'text';
  label: string;
};

type ButtonWithIcon = {
  variant: 'icon' | 'circle';
  icon: string;
};

type ButtonWithIconAndText = {
  variant: 'icon-and-text';
  icon: string;
  label: string;
  iconPosition: 'left' | 'right';
};

export type { ButtonProps };
