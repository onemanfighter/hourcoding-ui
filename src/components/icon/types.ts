export type IconProps = {} & (
  | {
      category: 'generic';
      size?: 'small' | 'medium' | 'large';
      variant?: 'filled' | 'outlined';
      color?: 'primary' | 'secondary' | 'tertiary';
      icon: GenericIcon;
    }
  | {
      category: 'common';
      size?: 'small' | 'medium' | 'large';
      color?: 'primary' | 'secondary' | 'tertiary';
      icon: CommonIcon;
    }
  | {
      category: 'hourcoding';
      size?: 'small' | 'medium' | 'large';
      color?: 'primary' | 'secondary' | 'tertiary';
      icon: HourcodingIcon;
    }
);

export enum HourcodingIcon {
  HOURCODING = 'hourCoding'
}
export enum GenericIcon {
  HOME = 'home',
  BACK = 'back'
}

export enum CommonIcon {
  ADD = 'add',
  REMOVE = 'remove',
  QR = 'qr'
}
