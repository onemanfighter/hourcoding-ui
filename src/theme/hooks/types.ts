import { HcColor } from '../types';

export type ButtonStyles = {
  backgroundColor?: HcColor;
  color: HcColor;
  borderColor?: HcColor;
};
export interface HcComponentStyles {
  Button: {
    contained: ButtonStyles;
    outlined: ButtonStyles;
    text: ButtonStyles;
    link: ButtonStyles;
  };
  Icon: {
    main: HcColor;
  };
}
