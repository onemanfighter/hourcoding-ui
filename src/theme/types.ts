import { blue, green, neutral, orange, red, system, yellow } from './Color';
import { TEXT_VARIANTS } from './TextVariants';

type Neutral = `neutral.${keyof typeof neutral}`;
type Blue = `blue.${keyof typeof blue}`;
type Green = `green.${keyof typeof green}`;
type Red = `red.${keyof typeof red}`;
type Orange = `orange.${keyof typeof orange}`;
type Yellow = `yellow.${keyof typeof yellow}`;
type System = `systemColor.${keyof typeof system}`;

export type HcColor = Neutral | Blue | Green | Red | Orange | Yellow | System;

// Text variants
export type TextVariant = {
  fontSize: number;
  fontWeight: string;
  lineHeight: number;
  textDecorationLine?: 'none' | 'underline';
};

type HeadersVariant = (typeof TEXT_VARIANTS.HEADERS)[keyof typeof TEXT_VARIANTS.HEADERS];
type TextContentVariant = (typeof TEXT_VARIANTS.CONTENT)[keyof typeof TEXT_VARIANTS.CONTENT];
type ButtonTextVariant = (typeof TEXT_VARIANTS.BUTTON.TEXT)[keyof typeof TEXT_VARIANTS.BUTTON.TEXT];

export type TextVariantsType = HeadersVariant | TextContentVariant | ButtonTextVariant | 'default';

// Theme type
export enum Theme {
  LIGHT = 'light',
  DARK = 'dark',
}

// Provider types
export type HCThemeProviderProps = {
  theme: Theme;
  children: React.ReactNode;
};
