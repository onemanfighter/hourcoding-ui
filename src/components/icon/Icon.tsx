import React from 'react';
import { IconProps } from './types';
import { ICON_LARGE_SIZE, ICON_MEDIUM_SIZE, ICON_SMALL_SIZE } from './constants';
import { iconsCommon, iconsFilled, iconsHourcoding, iconsOutlined } from './icons';

const Icon = ({ size, color, ...props }: IconProps) => {
  if (props.category === 'hourcoding') {
    return <img src={iconsHourcoding[props.icon]} alt="icon" />;
  }
  const iconSize = size
    ? size === 'small'
      ? ICON_SMALL_SIZE
      : size === 'medium'
        ? ICON_MEDIUM_SIZE
        : ICON_LARGE_SIZE
    : ICON_MEDIUM_SIZE;

  const iconSvg =
    props.category === 'common'
      ? iconsCommon[props.icon]
      : props.category === 'generic'
        ? props.variant === 'filled'
          ? iconsFilled[props.icon]
          : iconsOutlined[props.icon]
        : '';
  const markup = { __html: iconSvg };

  const id = `icon-${props.category}-${props.icon}`;
  return (
    <div
      id={id}
      style={{
        width: iconSize,
        height: iconSize,
        color: color === 'primary' ? 'blue' : color === 'secondary' ? 'red' : 'black'
      }}
      dangerouslySetInnerHTML={markup}
    />
  );
};

export default Icon;
