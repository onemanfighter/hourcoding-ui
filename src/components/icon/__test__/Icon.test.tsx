import React from 'react';
import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';
import { CommonIcon, GenericIcon } from '../types';
import Icon from '../Icon';

describe('Icon component', () => {
  const genericIcon = GenericIcon.HOME;
  const commonIcon = CommonIcon.ADD;

  describe('Generic Category', () => {
    it('icon component with generic category home with default props', () => {
      render(<Icon category="generic" icon={genericIcon} />);

      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with outline variant', () => {
      render(<Icon category="generic" icon={genericIcon} variant="outlined" />);

      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} variant="outlined" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with filled variant', () => {
      render(<Icon category="generic" icon={genericIcon} variant="filled" />);
      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} variant="filled" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with small size', () => {
      render(<Icon category="generic" icon={genericIcon} size="small" />);
      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} size="small" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with medium size', () => {
      render(<Icon category="generic" icon={genericIcon} size="medium" />);
      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} size="medium" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with large size', () => {
      render(<Icon category="generic" icon={genericIcon} size="large" />);
      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} size="large" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with primary color', () => {
      render(<Icon category="generic" icon={genericIcon} color="primary" />);
      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} color="primary" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with secondary color', () => {
      render(<Icon category="generic" icon={genericIcon} color="secondary" />);
      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} color="secondary" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Home icon with tertiary color', () => {
      render(<Icon category="generic" icon={genericIcon} color="tertiary" />);
      const icon = screen.getByTestId('icon-generic-home');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="generic" icon={genericIcon} color="tertiary" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });
  });

  describe('Common Category', () => {
    it('Add icon with default props', () => {
      render(<Icon category="common" icon={commonIcon} />);
      const icon = screen.getByTestId('icon-common-add');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer.create(<Icon category="common" icon={commonIcon} />).toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Add icon with small size', () => {
      render(<Icon category="common" icon={commonIcon} size="small" />);
      const icon = screen.getByTestId('icon-common-add');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="common" icon={commonIcon} size="small" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Add icon with medium size', () => {
      render(<Icon category="common" icon={commonIcon} size="medium" />);
      const icon = screen.getByTestId('icon-common-add');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="common" icon={commonIcon} size="medium" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Add icon with large size', () => {
      render(<Icon category="common" icon={commonIcon} size="large" />);
      const icon = screen.getByTestId('icon-common-add');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="common" icon={commonIcon} size="large" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Add icon with primary color', () => {
      render(<Icon category="common" icon={commonIcon} color="primary" />);
      const icon = screen.getByTestId('icon-common-add');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="common" icon={commonIcon} color="primary" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Add icon with secondary color', () => {
      render(<Icon category="common" icon={commonIcon} color="secondary" />);
      const icon = screen.getByTestId('icon-common-add');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="common" icon={commonIcon} color="secondary" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });

    it('Add icon with tertiary color', () => {
      render(<Icon category="common" icon={commonIcon} color="tertiary" />);
      const icon = screen.getByTestId('icon-common-add');
      expect(icon).toBeDefined();

      // Snapshot testing
      const componentTree = renderer
        .create(<Icon category="common" icon={commonIcon} color="tertiary" />)
        .toJSON();
      expect(componentTree).toMatchSnapshot();
    });
  });
});
