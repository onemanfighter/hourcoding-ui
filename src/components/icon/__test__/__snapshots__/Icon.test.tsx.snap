// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Common Category Snapshot tests Add icon with default props 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"M12 4.5v15m7.5-7.5h-15\\" />
</svg>
",
    }
  }
  id="icon-common-add"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Common Category Snapshot tests Add icon with large size 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"M12 4.5v15m7.5-7.5h-15\\" />
</svg>
",
    }
  }
  id="icon-common-add"
  style={
    Object {
      "color": "black",
      "height": 48,
      "width": 48,
    }
  }
/>
`;

exports[`Icon component Common Category Snapshot tests Add icon with medium size 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"M12 4.5v15m7.5-7.5h-15\\" />
</svg>
",
    }
  }
  id="icon-common-add"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Common Category Snapshot tests Add icon with primary color 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"M12 4.5v15m7.5-7.5h-15\\" />
</svg>
",
    }
  }
  id="icon-common-add"
  style={
    Object {
      "color": "blue",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Common Category Snapshot tests Add icon with secondary color 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"M12 4.5v15m7.5-7.5h-15\\" />
</svg>
",
    }
  }
  id="icon-common-add"
  style={
    Object {
      "color": "red",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Common Category Snapshot tests Add icon with small size 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"M12 4.5v15m7.5-7.5h-15\\" />
</svg>
",
    }
  }
  id="icon-common-add"
  style={
    Object {
      "color": "black",
      "height": 24,
      "width": 24,
    }
  }
/>
`;

exports[`Icon component Common Category Snapshot tests Add icon with tertiary color 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"M12 4.5v15m7.5-7.5h-15\\" />
</svg>
",
    }
  }
  id="icon-common-add"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with filled variant 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" fill=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path d=\\"M11.47 3.841a.75.75 0 0 1 1.06 0l8.69 8.69a.75.75 0 1 0 1.06-1.061l-8.689-8.69a2.25 2.25 0 0 0-3.182 0l-8.69 8.69a.75.75 0 1 0 1.061 1.06l8.69-8.689Z\\" />
  <path d=\\"m12 5.432 8.159 8.159c.03.03.06.058.091.086v6.198c0 1.035-.84 1.875-1.875 1.875H15a.75.75 0 0 1-.75-.75v-4.5a.75.75 0 0 0-.75-.75h-3a.75.75 0 0 0-.75.75V21a.75.75 0 0 1-.75.75H5.625a1.875 1.875 0 0 1-1.875-1.875v-6.198a2.29 2.29 0 0 0 .091-.086L12 5.432Z\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with large size 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "black",
      "height": 48,
      "width": 48,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with medium size 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with outline variant 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with primary color 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "blue",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with secondary color 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "red",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with small size 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "black",
      "height": 24,
      "width": 24,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test Home icon with tertiary color 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;

exports[`Icon component Generic Category snapshot test icon component with generic category home with default props 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\" strokeWidth={1.5} stroke=\\"currentColor\\" className=\\"w-6 h-6\\">
  <path strokeLinecap=\\"round\\" strokeLinejoin=\\"round\\" d=\\"m2.25 12 8.954-8.955c.44-.439 1.152-.439 1.591 0L21.75 12M4.5 9.75v10.125c0 .621.504 1.125 1.125 1.125H9.75v-4.875c0-.621.504-1.125 1.125-1.125h2.25c.621 0 1.125.504 1.125 1.125V21h4.125c.621 0 1.125-.504 1.125-1.125V9.75M8.25 21h8.25\\" />
</svg>
",
    }
  }
  id="icon-generic-home"
  style={
    Object {
      "color": "black",
      "height": 32,
      "width": 32,
    }
  }
/>
`;
